android/KMEA/app/src/main/java/com/tavultesoft/kmea/KMManager.java

```diff
@@ -2063,7 +2056,30 @@ public void run() {
 
           if (s.length() > 0) {
             SystemKeyboardShouldIgnoreSelectionChange = true;
-            ic.commitText(s, s.length());
+
+            // Using BreakIterator to count grapheme clusters
```

Comment:
I don't understand why we need to move the cursor at all? If we are deleting left and then inserting text, surely no cursor movement is required?

Reply:
I am concerned that using `BreakIterator` won't work if we are on a different version of Unicode than the target app. We shouldn't need to use `BreakIterator` at all: we should be able to do the calculation based purely on what changes we see happening in the text, shouldn't we?

Reply:
There's just no consistency when specifying lengths between Chrome and Firefox for left deletions and commit text.

For a context `s*😃` where the * is  cursor:
Originally on Chrome, we would commitText(p̂p̂, 4) because it's 4 code points. Chrome would then advance the cursor 4 places, causing it to overshoot.
With break iterator, I specify length of 2, but even then the cursor overshoots past the 😄 .

The cursor adjustment is similar to how we have to restore characters after a "deletion". Because the OS is inconsistent in the behavior.

Reply:
Using newCursorPosition `1` fixed the cursor position and also removed the use of BreakIterator
`ic.commitText(s, 1);`

Tested on Chrome and Firefox

Reply:
🥂